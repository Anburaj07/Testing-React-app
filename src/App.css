.App {
  text-align: center;
}

.size{
  width: 200px;
  height: 100px;  
}

.color{
  border: 1px solid rebeccapurple;
  color: cyan;
}