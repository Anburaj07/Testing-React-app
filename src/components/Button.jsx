import React from 'react'
import PropTypes from "prop-types"

const Button = ({children,size,color,func}) => {
  return (
    <button data-testid="test-button" className={`${size} ${color}`} onClick={func}>
        {children}
    </button>
  )
}

export default Button

Button.propTypes={
    children:PropTypes.string.isRequired,
    size:PropTypes.number.isRequired,
    color:PropTypes.string.isRequired
}