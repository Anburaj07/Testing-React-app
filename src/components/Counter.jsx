import React, { useState } from 'react'
import Button from './Button'

const Counter = () => {
    const [count,setCount]=useState(0)
    const handleAdd=()=>{
        setCount(prev=>prev+1);
    }
  return (
    <div data-testid="counter">
      <h1>Counter: {count}</h1>
      <Button func={handleAdd} color="gray" size="small">Add</Button>
    </div>
  )
}

export default Counter
