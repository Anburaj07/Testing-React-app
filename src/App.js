import './App.css';
import Button from './components/Button';
import Counter from './components/Counter';

function App() {
  return (
    <div className="App">
      {/* <Button >Click Me</Button>
       */}
       <Counter/>
    </div>
  );
}

export default App;
