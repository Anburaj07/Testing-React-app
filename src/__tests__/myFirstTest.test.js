import { fireEvent, render, screen } from "@testing-library/react"
import Button from "../components/Button"
import App from "../App";
import { click } from "@testing-library/user-event/dist/click";
import Counter from "../components/Counter";

//component testing
describe("Testing Button Component",()=>{
    it ("check Button is present on DOM or not",()=>{
    render(<Button>Testing Button</Button>)
    const button=screen.getByText("Testing Button");
    expect(button).toBeInTheDocument();
    });

    it("Button is rendered on App with text Click Me",()=>{
        render(<App/>)
        const button=screen.getAllByTestId("test-button");
        expect(button[0]).toBeInTheDocument();
        expect(button[0]).toHaveTextContent('Add')
    })

    it("OnClicking the button shold execute",()=>{
        const mockFun=jest.fn();
        render(<Button func={mockFun}>Click</Button>)
        const button=screen.getByTestId("test-button")
        fireEvent.click(button);
        fireEvent.click(button);
        // expect(mockFun).toBeCalled()
        expect(mockFun).toBeCalledTimes(2)
    })

    it("Should thorw error for color and size",()=>{
        jest.spyOn(console,"error")
        render(<Button color={1} size={false}>Click</Button>)
        expect(console.error).toBeCalledTimes(2)
    })
});

describe("Checking Counter function",()=>{
    it("Counter Should be present on DOM with initial value",()=>{
        render(<Counter/>)
        const counter=screen.getByTestId('counter')
        expect(counter).toBeInTheDocument();
        expect(counter).toHaveTextContent(0)
    })

    it("On Clicking add button counter should increase",()=>{
        render(<Counter/>);
        const counter=screen.getByTestId('counter')
        expect(counter).toHaveTextContent(0)
        const button=screen.getByTestId("test-button");
        fireEvent.click(button);
        expect(counter).toHaveTextContent(1);
        fireEvent.click(button);
        expect(counter).toHaveTextContent(2);
    })
})



