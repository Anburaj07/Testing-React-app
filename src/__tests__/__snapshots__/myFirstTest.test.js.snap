// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Counter function Button 1`] = `
<button
  className="large blue"
  data-testid="test-button"
/>
`;
